sympy/assumptions/cnf.py

```diff
@@ -0,0 +1,373 @@
+from itertools import combinations, product
+
+from sympy import S
+from sympy.logic.boolalg import BooleanFunction, Or, And, Not
+
+
+# Early encoding for literals
+literals_store = dict()
```

Comment:
This needs to be placed on the CNF object. As it is now, it is a global object that never gets cleared, which will be a memory leak for any long running processes that uses the new assumptions on a lot of SymPy expressions.